File: lambda/delete_exam_lambda.py
import json
import boto3
import os

dynamodb = boto3.resource('dynamodb')
table = dynamodb.Table('GeneratedExams')

def lambda_handler(event, context):
    try:
        body = event.get('body')
        if body and isinstance(body, str):
            body = json.loads(body)
        exam_id = body.get('examId') if body else None

        if not exam_id:
            return {
                "statusCode": 400,
                "body": json.dumps({"message": "Missing examId in request body"})
            }

        table.delete_item(Key={'examId': exam_id})

        return {
            "statusCode": 200,
            "body": json.dumps({"message": "Exam deleted successfully"})
        }
    except Exception as e:
        return {
            "statusCode": 500,
            "body": json.dumps({"message": "Error deleting exam", "error": str(e)})
        }

File: delete-exam-lambda.js
const AWS = require('aws-sdk');
const dynamoDb = new AWS.DynamoDB.DocumentClient();

exports.handler = async (event) => {
  try {
    const body = typeof event.body === 'string' ? JSON.parse(event.body) : event.body;
    const examId = body.examId;

    if (!examId) {
      return {
        statusCode: 400,
        body: JSON.stringify({ message: "Missing examId in request body" }),
      };
    }

    await dynamoDb.delete({
      TableName: 'Exams',
      Key: { examId }
    }).promise();

    return {
      statusCode: 200,
      body: JSON.stringify({ message: "Exam deleted successfully" }),
    };
  } catch (error) {
    return {
      statusCode: 500,
      body: JSON.stringify({ message: "Error deleting exam", error: error.message }),
    };
  }
};
